### DISCUSSION TITLE
black formater

### DISCUSSION THREAD

**Discussion by SebGue** [Eomys/pyleecan#577] [Q&A] [answered]
Hello @BonneelP 

what's your current black version. I'm asking since I got some issues with v22, i.e. some differences in format.
... and I also can't start black v20.8b1 for unknown reason.

Best regards, Sebastian

**Comment by BonneelP** [comment, marked as answer]
Hello,

I'm still using 20.8b1 but it is maybe time to update to use the latest version :) (hopefully format won't change at each version of black 🤞)
Maybe the error is the version of the package click ? I'm using 8.0.2

Best regards,
Pierre

**Comment by SebGue** [comment]
Hello, 

I never heard of package click, but downgrading to your version helped. Thank you!
Format changes I recognized so far with black v22 are removed white space in simple expressions with powers
`a**2 + b**2` (v20: `a ** 2 + b ** 2`) and removed 'u' in u-strings (e.g. in class property docs).

Best regards, Sebastian

**Comment by SebGue** [comment]
Hello @BonneelP ,

what is your current version of black? Regardless, if I use v20 or current version, I get changes in almost every class.
Do you think we should upgrade? Or should I try to find an appropriate version?

BR Sebastian

**Comment by BonneelP** [comment]
Hello Sebastian, 

Yes I think it is more than time that we update the black version. We can move to the latest version (24.4.2) and hope that new black release won't affect too much the formatting.

Best regards,
Pierre

**Reply by SebGue** [reply]
@BonneelP 

Okay, then I can do a PR with respective changes.

OT: You meantioned Linux compatibility in the other issue. Actually I try to add full Elmer support within the next weeks. So maybe this is an alternative to FEMM.

BR Sebastian

**Reply by BonneelP** [reply]
Thank you for the black PR :)

That would a great step forward for pyleecan, thank you for that :)

**Reply by SebGue** [reply]
@BonneelP Regarding black see #696
